packages/eas-cli/src/credentials/ios/appstore/bundleIdCapabilities.ts

```diff
@@ -0,0 +1,415 @@
+  const disabled: string[] = [];
+  const request: { capabilityType: CapabilityType; option: any }[] = [];
+  for (const [key, value] of Object.entries(entitlements)) {
+    if (value) {
+      const capability = CapabilityMapping.find(capability => capability.entitlement === key);
+      if (capability) {
+        const existingIndex = capabilities.findIndex(existing =>
+          existing.isType(capability.capability)
+        );
+        const existing = existingIndex > -1 ? capabilities[existingIndex] : null;
+        // Only skip if the existing capability is a simple boolean value,
+        // if it has more complex settings then we should always update it.
+        if (existing && existing?.attributes.settings == null) {
+          // Remove the item from the list of capabilities so we don't disable it.
+          capabilities.splice(existingIndex, 1);
+          if (Log.isDebug) {
+            Log.log(`Skipping existing capability: ${key} (${capability.name})`);
+          }
+          continue;
+        }
+
+        if (!capability.validateOptions(value)) {
+          throw new Error(`iOS entitlement "${key}" has invalid value "${value}".`);
+        }
+        enabled.push(capability.name);
+
+        const option = capability.getOptions(value, entitlements);
+
+        request.push({
+          capabilityType: capability.capability,
+          option,
+        });
+      } else {
+        if (Log.isDebug) {
+          Log.log(`Skipping unhandled entitlement: ${key}`);
+        }
+      }
+    }
+  }
+
+  // Disable any extras that aren't present, this functionality is kinda unreliable because managed apps
+  // might be enabling capabilities in modifiers.
+
+  for (const existing of capabilities) {
+    // Special case APNS because it's always enabled in Expo,
+    // GC and IAP are always enabled in apps by default so we should avoid modifying them.
+    if (
+      existing.isType(CapabilityType.IN_APP_PURCHASE) ||
+      existing.isType(CapabilityType.PUSH_NOTIFICATIONS) ||
+      existing.isType(CapabilityType.GAME_CENTER)
+    ) {
+      continue;
+    }
+
+    if (existing.attributes) {
+      let adjustedType: string | undefined = existing.attributes.capabilityType;
+      if (!adjustedType) {
+        adjustedType = existing.id.replace(`${bundleId.id}_`, '');
+      }
+
+      // Only disable attributes that we handle,
+      // this enables devs to turn on capabilities outside of EAS without worrying about us disabling them.
+      const capability = CapabilityMapping.find(
+        capability => capability.capability === adjustedType
+      );
+      if (
+        !!capability &&
+        !request.find(request => request.capabilityType && existing.isType(request.capabilityType))
+      ) {
+        request.push({
+          // @ts-ignore
+          capabilityType: adjustedType,
+          option: CapabilityTypeOption.OFF,
+        });
+
+        disabled.push(capability?.name || adjustedType);
```

Comment:
```suggestion
        disabled.push(capability.name);
```
`?` (and the fallback value) shouldn't be necessary if you apply the previous suggestion.